src/onegov/org/app.py

```diff
@@ -193,23 +193,25 @@ def ticket_permissions(self):
         return result
 
     @orm_cached(policy='on-table-change:pages')
-    def homepage_pages(self) -> dict[int, list[Topic]]:
-
+    def topics_cached(self) -> list[tuple[int, 'Topic']]:
         def visit_topics(
-            pages: 'Iterable[Page]',
-            root_id: int | None = None
+            pages: 'Iterable[Page]', root_id: int | None = None
         ) -> 'Iterator[tuple[int, Topic]]':
             for page in pages:
+
                 if isinstance(page, Topic):
                     yield root_id or page.id, page
 
                 yield from visit_topics(
                     page.children,
-                    root_id=root_id or page.id
+                    root_id=root_id
```

Comment:
This will not work correctly, since the widget for homepage_pages expects all the nested topics to exist in one combined flat list per root page.

The `or page.id` is necessary to forward the root_id to the children tree. So either leave it like it was or use a new variable `current_root_id = root_id or page.id` and use it for the `yield` above and down here, if you want to avoid the duplicated code.